lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -541,44 +571,52 @@ def __init__(self, module):
         self.cert = None
         self.privatekey = None
         self.csr = None
+        self.backend = backend
         self.module = module
 
     def get_relative_time_option(self, input_string, input_name):
         """Return an ASN1 formatted string if a relative timespec
            or an ASN1 formatted string is provided."""
         result = input_string
         if result.startswith("+") or result.startswith("-"):
-            result = crypto_utils.convert_relative_to_datetime(
-                result).strftime("%Y%m%d%H%M%SZ")
+            result_datetime = crypto_utils.convert_relative_to_datetime(
+                result)
+            if self.backend == 'pyopenssl':
+                return result_datetime.strftime("%Y%m%d%H%M%SZ")
+            elif self.backend == 'cryptography':
+                return result_datetime
         if result is None:
             raise CertificateError(
                 'The timespec "%s" for %s is not valid' %
                 input_string, input_name)
+        if self.backend == 'cryptography':
+            result = datetime.datetime.strptime(input_string, '%Y%m%d%H%M%SZ')
```

Comment:
There are some more possible input formats for ASN.1 TIME, see here: https://www.obj-sys.com/asn1tutorial/node15.html

Reply:
Added a try/except for the various combinations of ASN.1 TIME